src/d3d9/d3d9_device.h

```diff
@@ -1186,6 +1186,13 @@ namespace dxvk {
         : GetHelper(m_state.psConsts);
     }
 
+    inline uint32_t CalculateVertexSizeUP(uint32_t vertexCount, uint32_t stride) {
+      if (unlikely(!vertexCount))
```

Comment:
can we exit early from `DrawPrimitiveUP` instead if the app is dumb? We shouldn't submit an empty draw to the backend.

Reply:
Done